### DISCUSSION TITLE
How to set memory usage limit in pytest?

### DISCUSSION THREAD

**Discussion by Coder2023Coder** [pytest-dev/pytest#11278] [Q&A]
Since the space complexity of some functions I am testing are high and the operating system kills their testing execution, I want to set a memory usage limit in a way that as soon as a test case running the main code under test, exceeds taking that allowed amount of memory, pytest stops it and flags it as failed. I am NOT using command line so I am looking for a solution to implement in my code. 

I did the following way but it didn't work:

```
import pytest
import resource

@pytest.fixture(autouse=True)
def set_resource_limits():
    soft, hard = resource.getrlimit(resource.RLIMIT_AS)
    resource.setrlimit(resource.RLIMIT_AS, (10 * 1024 * 1024, hard))  # 10 MB limit
```

My pytest setup is as follows:
```
pytest_params = [ "-v", "tests.py", "-c", "my_ini.ini"]
pytest.main(pytest_params)
```

**Comment by nicoddemus** [comment]
Hi @Coder2023Coder,

I think you will need to follow the same approach used by `pytest-timeout`, where your condition to "stop the test and dump a traceback" is not a maximum elapsed time, but a maximum amount of memory consumed by the test. I would fork pytest-timeout and play around with its code.

**Reply by Coder2023Coder** [reply]
Can you share the link of `pytest-timeout`, please?

Also, what is wrong with the code snippet I shared?

**Reply by nicoddemus** [reply]
> Can you share the link of pytest-timeout, please?

You can easily find it on Google.

> Also, what is wrong with the code snippet I shared?

Don't know how `resource` works, but pointed to something would provide a solution where you can stop a test when it uses too much memory.
